x/params/keeper.go

```diff
@@ -0,0 +1,328 @@
+	return
+}
+
+// GetInt is helper function for sdk.Int params
+func (k Getter) GetInt(ctx sdk.Context, key string) (res sdk.Int, err error) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err = k.k.cdc.UnmarshalBinary(bz, &res)
+	return
+}
+
+// GetUint is helper function for sdk.Uint params
+func (k Getter) GetUint(ctx sdk.Context, key string) (res sdk.Uint, err error) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err = k.k.cdc.UnmarshalBinary(bz, &res)
+	return
+}
+
+// GetRat is helper function for rat params
+func (k Getter) GetRat(ctx sdk.Context, key string) (res sdk.Rat, err error) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err = k.k.cdc.UnmarshalBinary(bz, &res)
+	return
+}
+
+// GetStringWithDefault is helper function for string params with default value
+func (k Getter) GetStringWithDefault(ctx sdk.Context, key string, def string) (res string) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetBoolWithDefault is helper function for bool params with default value
+func (k Getter) GetBoolWithDefault(ctx sdk.Context, key string, def bool) (res bool) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetInt16WithDefault is helper function for int16 params with default value
+func (k Getter) GetInt16WithDefault(ctx sdk.Context, key string, def int16) (res int16) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetInt32WithDefault is helper function for int32 params with default value
+func (k Getter) GetInt32WithDefault(ctx sdk.Context, key string, def int32) (res int32) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetInt64WithDefault is helper function for int64 params with default value
+func (k Getter) GetInt64WithDefault(ctx sdk.Context, key string, def int64) (res int64) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetUint16WithDefault is helper function for uint16 params with default value
+func (k Getter) GetUint16WithDefault(ctx sdk.Context, key string, def uint16) (res uint16) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
+	if err != nil {
+		return def
+	}
+	return
+}
+
+// GetUint32WithDefault is helper function for uint32 params with default value
+func (k Getter) GetUint32WithDefault(ctx sdk.Context, key string, def uint32) (res uint32) {
+	store := ctx.KVStore(k.k.key)
+	bz := store.Get([]byte(key))
+	err := k.k.cdc.UnmarshalBinary(bz, &res)
```

Comment:
I don't think this (and all similar) check what we want. I think we want to return the default if there is *no* set value, and probably still panic on an unmarshal error - we should never set a non-unmarshalable value - thoughts?